pkg/apis/core/types.go

```diff
@@ -5732,6 +5732,14 @@ type TopologySpreadConstraint struct {
 	// This is a alpha-level feature enabled by the NodeInclusionPolicyInPodTopologySpread feature flag.
 	// +optional
 	NodeTaintsPolicy *NodeInclusionPolicy
+	// MatchLabelKeys is a set of pod label keys to select the pods over which
+	// spreading will be calculated. The keys are used to lookup values from the
+	// incoming pod labels, those key-value labels are ANDed with `LabelSelector`
+	// to select the group of existing pods over which spreading will be calculated
+	// for the incoming pod. Keys that don't exist in the incoming pod labels will
+	// be ignored.
+	// +optional
+	MatchLabelKeys []string
```

Comment:
Since the rolling update not working as expected, shouldn't we add some keys like `pod-template-hash` as default values. Yes it will change the current behavior, but considering we are recreating pods, it's actually what end-users wants.

Reply:
Can be considered in a later version maybe beta.

Reply:
Then from a different perspective, we may treat this as providing a configurable way for end-users to prevent the unexpected results without breaking current behaviors. (Still weird for I would like to treat this as a bug.)

Besides, IIRC, we'll meet this problem the same with podAffinity/podAntiAffinity, that's part of the reason why descheduler incubated. right? If so, we should look for a mechanism cover these scenarios consistently, about distinguishing between current and rolling-out versions. This is somehow outside the scope of scheduling, but related.